lib/Elastica/Query/SpanFirst.php

```diff
@@ -0,0 +1,76 @@
+<?php
+namespace Elastica\Query;
+
+use Elastica\Exception\InvalidException;
+
+/**
+ * SpanFirst query.
+ *
+ * @author Alessandro Chitolina <alekitto@gmail.com>
+ *
+ * @link https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-span-first-query.html
+ */
+class SpanFirst extends AbstractSpanQuery
+{
+    /**
+     * Constructs a SpanFirst query object.
+     *
+     * @param \Elastica\Query\AbstractQuery|array $match OPTIONAL
+     * @param int $end OPTIONAL
+     */
+    public function __construct($match = null, $end = null)
+    {
+        if (null !== $match) {
+            $this->setMatch($match);
+        }
+
+        if (null !== $end) {
+            $this->setEnd($end);
+        }
+    }
+
+    /**
+     * Set the query to be wrapped into the span multi query.
+     *
+     * @param \Elastica\Query\AbstractSpanQuery|array $args Matching query
+     *
+     * @return $this
+     */
+    public function setMatch($args)
+    {
+        return $this->_addQuery('match', $args);
+    }
+
+    /**
+     * Set the maximum end position for the match query.
+     *
+     * @param int $end
+     *
+     * @return $this
+     */
+    public function setEnd($end)
+    {
+        $this->setParam('end', $end);
+
+        return $this;
+    }
+
+    /**
+     * Adds a query to the current object.
+     *
+     * @param string                              $type Query type
+     * @param \Elastica\Query\AbstractQuery|array $args Query
+     *
+     * @throws \Elastica\Exception\InvalidException If not valid query
+     *
+     * @return $this
+     */
+    protected function _addQuery($type, $args)
```

Comment:
This more looks like `_setQuery` as in case a second query is set, it will overwrite the previous one.